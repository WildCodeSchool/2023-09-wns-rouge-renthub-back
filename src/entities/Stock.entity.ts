import {
  Column,
  Entity,
  PrimaryGeneratedColumn,
  JoinColumn,
  ManyToOne,
} from 'typeorm'
import { IsOptional, Length, IsDate } from 'class-validator'
import { Field, ID, InputType, ObjectType } from 'type-graphql'
import { ProductReference } from './ProductReference.entity'
import { EntityWithDefault } from './EntityWithDefault'
import { ObjectId } from './ObjectId'

@Entity()
@ObjectType()
export class Stock extends EntityWithDefault {
  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id!: number

  @Column({ length: 50, nullable: true })
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Field()
  name!: string

  @Column({ default: true })
  @IsOptional()
  @Field({ nullable: true })
  isAvailable!: boolean

  @Column({ length: 50 })
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Field()
  serialNumber!: string

  @Column({ type: 'timestamp with time zone' })
  @IsOptional()
  @IsDate({ message: 'Doit être une date valide' })
  @Field(() => Date, { nullable: true })
  purchaseDataTime!: Date

  @Column({ length: 50, nullable: true })
  @IsOptional()
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Field({ nullable: true })
  supplier!: string

  @Column({ length: 50, nullable: true })
  @IsOptional()
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Field({ nullable: true })
  sku!: string

  @ManyToOne(
    () => ProductReference,
    (productReference) => productReference.stock
  )
  @JoinColumn({ name: 'productReference' })
  @Field(() => ProductReference)
  productReference!: ProductReference
}

@InputType()
export class StockCreateInput {
  @Field()
  name: string

  @Field({ nullable: true })
  isAvailable: boolean

  @Field()
  serialNumber: string

  @Field(() => Date, { nullable: true })
  purchaseDataTime: Date

  @Field({ nullable: true })
  supplier?: string

  @Field({ nullable: true })
  sku: string

  @Field()
  productReference: ObjectId
}

@InputType()
export class StockUpdateInput {
  @Field(() => ID)
  id: number

  @Field({ nullable: true })
  name: string

  @Field({ nullable: true })
  isAvailable: boolean

  @Field({ nullable: true })
  serialNumber: string

  @Field(() => Date, { nullable: true })
  purchaseDataTime: Date

  @Field({ nullable: true })
  supplier?: string

  @Field({ nullable: true })
  sku: string

  @Field({ nullable: true })
  productReference: ObjectId
}
