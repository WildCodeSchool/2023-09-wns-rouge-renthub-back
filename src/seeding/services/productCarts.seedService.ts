import { SeederFactoryManager } from 'typeorm-extension'
import { ProductCart } from '../../entities/ProductCart.entity'
import { ProductReference } from '../../entities/ProductReference.entity'
import { Cart } from '../../entities/Cart.entity'

export type ProductCartsSeederTypes = {
  productCartsSaved: ProductCart[]
}

/**
 * Seed the product carts by associating random products with each cart.
 * @param cartsSaved - The array of saved carts.
 * @param productReferences - The array of product references.
 * @param factoryManager - The seeder factory manager.
 * @returns An object containing the saved product carts.
 */
export default async function productCartsSeeder(
  cartsSaved: Cart[],
  productReferences: ProductReference[],
  factoryManager: SeederFactoryManager
) {
  const productCartsSaved: ProductCart[] = []

  for (const cart of cartsSaved) {
    const productCartFactory = factoryManager.get(ProductCart)
    const randomCountProducts = Math.floor(Math.random() * 5)

    const productReferencesToSave: ProductReference[] = []

    for (let i = 0; i < randomCountProducts; i++) {
      const listOfProductsReferencesLeft = productReferences.filter(
        (el) => !productReferencesToSave.find((el2) => el.id === el2.id)
      )
      const productReference =
        listOfProductsReferencesLeft[
          Math.floor(Math.random() * listOfProductsReferencesLeft.length)
        ]
      productReferencesToSave.push(productReference)

      const productCart = await productCartFactory.make()
      productCart.quantity = Math.ceil(Math.random() * 5)

      productCart.productReference =
        productReferencesToSave[productReferencesToSave.length - 1]

      productCart.cartReference = cart

      await productCart.save()
      productCartsSaved.push(productCart)
    }
  }
  return { productCartsSaved }
}
