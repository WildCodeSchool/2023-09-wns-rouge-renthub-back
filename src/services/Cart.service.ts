import { Repository } from 'typeorm'
import { dataSource } from '../datasource'
import { Cart } from '../entities/Cart.entity'
import { calculateDaysBetweenDates } from '../utils/utils'

export class CartService {
  db: Repository<Cart>
  constructor() {
    this.db = dataSource.getRepository(Cart)
  }

  public static calculateTotalPrice(cart: Cart) {
    let totalPrice = 0

    for (const productCart of cart.productCarts) {
      const daysCount = calculateDaysBetweenDates(
        productCart.dateTimeStart,
        productCart.dateTimeEnd
      )
      totalPrice +=
        productCart.quantity * productCart.productReference.price * daysCount
    }

    return totalPrice
  }

  async find(id: number) {
    const cart = await this.db.findOne({
      where: { id },
      relations: {
        owner: true,
        productCarts: { productReference: { category: true } },
      },
    })
    if (!cart) throw new Error('Cart not found')

    const totalPrice = CartService.calculateTotalPrice(cart)
    if (cart.totalPrice != totalPrice) {
      cart.totalPrice = totalPrice
      await cart.save()
    }
    return cart
  }
}
