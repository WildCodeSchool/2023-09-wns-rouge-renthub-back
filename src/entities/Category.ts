import 'reflect-metadata'
import {
  BeforeInsert,
  BeforeUpdate,
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToMany,
  OneToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import { Field, ID, InputType, ObjectType } from 'type-graphql'
import { Picture } from './Picture'

@Entity()
@ObjectType()
export class Category {
  @BeforeInsert()
  updateDatesOnInsert() {
    this.createdAt = new Date()
    this.updatedAt = new Date()
  }

  @BeforeUpdate()
  updateDatesOnUpdate() {
    this.updatedAt = new Date()
  }

  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id: number

  @Column({ length: 50, unique: true })
  @Field()
  name: string

  // index permet ordonner les catÃ©gories pour l'affichage
  @Column({})
  @Field(() => ID)
  index: number

  @Column({ default: true })
  @Field(() => Boolean, { nullable: true })
  display: boolean

  @Column({ nullable: true })
  @Field(() => ID, { nullable: true })
  createdBy: number

  @Column({ nullable: true })
  @Field(() => ID, { nullable: true })
  updatedBy: number

  @CreateDateColumn({ type: 'timestamp with time zone' })
  @Field()
  createdAt: Date

  @UpdateDateColumn({ type: 'timestamp with time zone', nullable: true })
  @Field({ nullable: true })
  updatedAt!: Date

  @ManyToOne(() => Category, (category) => category.childCategories, {
    nullable: true,
  })
  @Field(() => Category, { nullable: true })
  parentCategory?: Category

  @OneToMany(() => Category, (category) => category.parentCategory, {
    cascade: true,
  })
  @Field(() => [Category], { nullable: true })
  childCategories?: Category[]

  @OneToOne(() => Picture, { nullable: true })
  @JoinColumn({ name: 'pictureId', referencedColumnName: 'id' })
  @Field(() => Picture, { nullable: true })
  picture?: Picture
}

@InputType()
export class CategoryCreateInput {
  @Field()
  name: string

  @Field(() => ID)
  index: number

  @Field(() => Boolean, { nullable: true })
  display: boolean

  @Field(() => ID)
  createdBy: number

  @Field(() => ID, { nullable: true })
  parentCategoryId: number

  // @Field(() => PictureCreateInput, { nullable: true })
  // picture?: PictureCreateInput
}

@InputType()
export class CategoryUpdateInput {
  @Field(() => ID)
  id: number

  @Field({ nullable: true })
  name?: string

  @Field(() => ID, { nullable: true })
  index?: number

  @Field(() => Boolean, { nullable: true })
  display?: boolean

  @Field(() => ID)
  updatedBy: number

  @Field(() => ID, { nullable: true })
  parentCategoryId: number
}
