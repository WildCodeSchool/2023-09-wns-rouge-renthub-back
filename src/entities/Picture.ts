import {
  BaseEntity,
  BeforeInsert,
  BeforeUpdate,
  Column,
  CreateDateColumn,
  Entity,
  OneToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import { Field, ID, InputType, ObjectType } from 'type-graphql'
import { Category } from './Category'

@Entity()
@ObjectType()
export class Picture extends BaseEntity {
  @BeforeInsert()
  updateDatesOnInsert() {
    this.createdAt = new Date()
    this.updatedAt = new Date()
  }

  @BeforeUpdate()
  updateDatesOnUpdate() {
    this.updatedAt = new Date()
  }

  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id: number

  @Column()
  @Field()
  filename: string

  @Column({ nullable: true })
  @Field({ nullable: true })
  urlHD: string

  @Column({ nullable: true })
  @Field({ nullable: true })
  urlMiniature: string

  @Column()
  @Field(() => ID)
  createdBy: number

  @Column({ nullable: true })
  @Field(() => ID, { nullable: true })
  updatedBy: number

  @CreateDateColumn({ type: 'timestamp with time zone' })
  @Field()
  createdAt: Date

  @UpdateDateColumn({ type: 'timestamp with time zone', nullable: true })
  @Field({ nullable: true })
  updatedAt?: Date

  @OneToOne(() => Category, (category) => category.picture)
  @Field(() => Category, { nullable: true })
  category: Category
}

@InputType()
export class PictureCreateInput {
  @Field()
  filename: string

  @Field()
  urlHD: string

  @Field({ nullable: true })
  urlMiniature: string
}

@InputType()
export class PictureUpdate {
  @Field({ nullable: true })
  filename: string

  @Field({ nullable: true })
  urlHD: string

  @Field({ nullable: true })
  urlMiniature: string
}
