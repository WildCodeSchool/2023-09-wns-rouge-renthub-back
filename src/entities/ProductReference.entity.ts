import {
  Column,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm'
import { Field, ID, InputType, Int, ObjectType } from 'type-graphql'
import { Category } from './Category.entity'
import { PictureProduct } from './PictureProduct.entity'
import { IsBoolean, Length } from 'class-validator'
import { ObjectId } from './ObjectId'
import { EntityWithDefault } from './EntityWithDefault'
import { Stock } from './Stock.entity'
import { ProductCart } from './ProductCart.entity'

@Entity()
@ObjectType()
export class ProductReference extends EntityWithDefault {
  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id!: number

  @Column({ length: 150 })
  @Length(2, 150)
  @Field()
  name!: string

  @Column({ type: 'text' })
  @Length(10, 5000)
  @Field()
  description!: string

  @Column()
  @Field(() => Int)
  index!: number

  @Column({ default: true })
  @IsBoolean()
  @Field(() => Boolean)
  display: boolean

  @Column({ length: 150 })
  @Length(2, 150)
  @Field()
  brandName!: string

  @Column()
  @Field(() => Int)
  price!: number

  @ManyToOne(() => Category, (category) => category.productReferences)
  @Field(() => Category)
  category?: Category

  @OneToMany(
    () => PictureProduct,
    (pictureProduct) => pictureProduct.productReference,
    { cascade: true }
  )
  @Field(() => [PictureProduct], { nullable: true })
  pictureProduct!: PictureProduct[]

  @OneToMany(() => Stock, (stock) => stock.productReference, {
    cascade: true,
    onDelete: 'CASCADE',
  })
  @Field(() => [Stock], { nullable: true })
  stock!: Stock

  @OneToMany(() => ProductCart, (productCart) => productCart.productReference, {
    cascade: true,
  })
  @Field(() => [ProductCart])
<<<<<<< HEAD
  productCarts!: ProductCart[]
=======
  productCart!: ProductCart[]
>>>>>>> 298666a384255480e5436e0df250a8ecc63dd327
}

@InputType()
export class ProductReferenceCreateInput {
  @Field()
  name: string

  @Field()
  description: string

  @Field(() => Int)
  index: number

  @Field(() => Boolean, { nullable: true })
  display: boolean

  @Field({ nullable: true })
  brandName?: string

  @Field(() => Int)
  price: number

  @Field()
  createdBy: ObjectId

  @Field()
  category: ObjectId
}

@InputType()
export class ProductReferenceUpdateInput {
  @Field({ nullable: true })
  updatedBy: ObjectId

  @Field({ nullable: true })
<<<<<<< HEAD
  name: string

  @Field({ nullable: true })
  description: string

  @Field(() => Int, { nullable: true })
  index: number
=======
  name?: string

  @Field({ nullable: true })
  description?: string

  @Field(() => Int, { nullable: true })
  index?: number
>>>>>>> 298666a384255480e5436e0df250a8ecc63dd327

  @Field(() => Boolean, { nullable: true })
  display?: boolean

  @Field({ nullable: true })
  brandName?: string

  @Field(() => Int, { nullable: true })
<<<<<<< HEAD
  price: number

  @Field({ nullable: true })
  category: ObjectId
=======
  price?: number

  @Field({ nullable: true })
  category?: ObjectId
>>>>>>> 298666a384255480e5436e0df250a8ecc63dd327
}
