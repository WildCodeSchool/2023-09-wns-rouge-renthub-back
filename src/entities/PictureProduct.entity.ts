import {
  BaseEntity,
  Column,
  Entity,
  ManyToOne,
  PrimaryGeneratedColumn,
} from 'typeorm'
import { Field, ID, Int, ObjectType } from 'type-graphql'
import { Picture } from './Picture.entity'
import { ProductReference } from './ProductReference.entity'

@Entity()
@ObjectType()
export class PictureProduct extends BaseEntity {
  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id!: number

  @Column()
  @Field(() => Int)
  index!: number

  @ManyToOne(() => Picture, (picture) => picture.pictureProduct)
  @Field(() => Picture, { nullable: true })
  picture!: Picture

  @ManyToOne(
    () => ProductReference,
    (productReference) => productReference.pictureProduct
  )
  @Field(() => ProductReference, { nullable: true })
  productReference!: ProductReference
}
