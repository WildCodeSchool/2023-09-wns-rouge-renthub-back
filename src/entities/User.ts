import {
  BaseEntity,
  Column,
  Entity,
  PrimaryGeneratedColumn,
  OneToOne,
  JoinColumn,
  ManyToOne,
  UpdateDateColumn,
  CreateDateColumn,
  OneToMany,
} from 'typeorm'
import {
  IsEmail,
  IsNumberString,
  IsOptional,
  Length,
  Matches,
  IsDate,
} from 'class-validator'
import { Field, ID, InputType, ObjectType } from 'type-graphql'
import { Picture } from './Picture'
import { Role } from './Role'
import { ObjectId } from './ObjectId'
import { VerificationCode } from './VerificationCode'

@Entity()
@ObjectType()
export class User extends BaseEntity {
  @PrimaryGeneratedColumn()
  @Field(() => ID)
  id!: number

  @Column({ length: 50, nullable: true })
  @IsOptional()
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Matches(/^[a-zA-ZÀ-ÿ-]+$/, {
    message: 'Le prénom ne doit contenir que des lettres',
  })
  @Field({ nullable: true })
  firstName!: string

  @Column({ length: 50, nullable: true })
  @IsOptional()
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Matches(/^[a-zA-ZÀ-ÿ-]+$/, {
    message: 'Le nom de famille ne doit contenir que des lettres',
  })
  @Field({ nullable: true })
  lastName!: string

  @Column({ length: 50, nullable: true })
  @IsOptional()
  @Length(2, 50, { message: 'Entre 2 et 50 caractères' })
  @Field({ nullable: true })
  nickName!: string

  @Column({ type: 'date', nullable: true })
  @IsOptional()
  @IsDate({ message: 'Doit être une date valide' })
  @Field(() => Date, { nullable: true })
  dateOfBirth!: Date

  @Column({ length: 250 })
  hashedPassword!: string

  @Column({ length: 10, nullable: true })
  @IsOptional()
  @IsNumberString(
    {},
    { message: 'Le numéro de téléphone doit être une chaîne de chiffres' }
  )
  @Length(10, 10, {
    message: 'Le numéro de téléphone doit avoir exactement 10 chiffres',
  })
  @Field({ nullable: true })
  phoneNumber!: string

  @Column({ length: 255, unique: true })
  @Field()
  @IsEmail()
  email!: string

  @Column({ default: false })
  @Field()
  isVerified!: boolean

  @Column({ type: 'timestamp', nullable: true })
  @Field(() => Date, { nullable: true })
  lastConnectionDate!: Date

  @CreateDateColumn({ type: 'timestamp' })
  @Field(() => Date)
  createdAt!: Date

  @ManyToOne(() => User, (user) => user.createdBy)
  @Field(() => User, { nullable: true })
  createdBy!: User

  @UpdateDateColumn()
  @Field(() => Date)
  updatedAt!: Date

  @ManyToOne(() => User, (user) => user.updatedBy, { nullable: true })
  @Field(() => User, { nullable: true })
  updatedBy!: User

  @OneToOne(() => Picture, { nullable: true })
  @IsOptional()
  @JoinColumn()
  @Field({ nullable: true })
  picture?: Picture

  //TODO : comment this line because
  @OneToMany(
    () => VerificationCode,
    (verificationCode) => verificationCode.user,
    { cascade: true }
  )
  verificationCodes!: VerificationCode[]

  @ManyToOne(() => Role, (role) => role.user)
  @JoinColumn({ name: 'role' })
  @Field(() => Role, { nullable: true })
  role!: Role
}

@InputType()
export class UserCreateInput {
  @Field({ nullable: true })
  firstName!: string

  @Field({ nullable: true })
  lastName!: string

  @Field({ nullable: true })
  nickName!: string

  @Field({ nullable: true })
  dateOfBirth!: string

  @Field()
  @Matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d!@#$%^&*()_+]{8,}$/, {
    message:
      'Password is not valid. At least 8 characters, 1 uppercase, 1 lowercase, 1 special characters and 1 number required!',
  })
  password!: string

  @Field({ nullable: true })
  phoneNumber?: string

  @Field()
  @Matches(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/, {
    message: 'Email is not valid!',
  })
  email!: string

  @Field({ nullable: true })
  pictureId?: ObjectId
}

@InputType()
export class UserUpdateInput {
  @Field({ nullable: true })
  firstName!: string

  @Field({ nullable: true })
  lastName!: string

  @Field({ nullable: true })
  nickName!: string

  @Field({ nullable: true })
  dateOfBirth!: Date

  @Field({ nullable: true })
  @Matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/)
  password!: string

  @Field({ nullable: true })
  phoneNumber!: string

  @Field({ nullable: true })
  isVerified!: boolean

  @Field(() => ID, { nullable: true })
  updatedBy?: number

  @Field({ nullable: true })
  pictureId?: number
}

@InputType()
export class UserLoginInput {
  @Field()
  email!: string

  @Field()
  password!: string
}

@ObjectType()
export class UserContext {
  @Field()
  id!: number

  @Field()
  nickName!: string

  @Field()
  picture!: string
}

@ObjectType()
export class VerifyEmailResponse {
  @Field()
  success!: boolean

  @Field({ nullable: true })
  message?: string
}
