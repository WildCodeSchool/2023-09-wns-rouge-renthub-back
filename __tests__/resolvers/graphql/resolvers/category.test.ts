import { buildSchema } from 'type-graphql'
import { graphql } from 'graphql'
import { CategoriesResolver } from '../../../../src/resolvers/Category.resolver'
import { CategoryService } from '../../../../src/services/Category.services'
import { describe, expect, it, jest } from '@jest/globals'
import { Category } from '../../../../src/entities/Category'

describe('Categories Resolver - Snapshot', () => {
  it('should match snapshot', async () => {
    // Créer une  de service
    const service = new CategoryService()

    // Espionner la méthode list pour renvoyer des données fictives
    jest.spyOn(service, 'list').mockImplementation(async () => {
      return [
        {
          id: 1,
          name: 'Category 1',
          index: 1,
          display: true,
          createdBy: 'admin',
          updatedBy: 'admin',
          createdAt: new Date(),
          updatedAt: new Date(),
          parentCategory: null, // Mettez la référence parente si nécessaire
          childCategories: [], // Ajoutez des catégories enfants si nécessaire
          picture: null, // Ajoutez une référence à l'image si nécessaires
          updateDatesOnInsert: jest.fn(),
          updateDatesOnUpdate: jest.fn(),
        } as unknown as Category,
      ]
    })
    // Construire le schéma GraphQL avec les résolveurs
    const schema = await buildSchema({
      resolvers: [CategoriesResolver],
      validate: true, // Désactiver la validation du schéma pour éviter les erreurs pendant les tests
    })

    // Exécuter la requête GraphQL
    const res = await graphql({
      schema,
      source: `
        query {
          listCategories {
            id
            name
            index
            display
            createdBy
            updatedBy
            createdAt
            updatedAt
            parentCategory {
              name
            }
            childCategories {
              name
            }
            picture {
              urlHD
            }
          }
        }
      `,
    })

    // Vérifier si les données renvoyées correspondent au snapshot
    expect(res).toMatchSnapshot()
  })

  // it('can get a list of categories', async () => {
  //   const catService = new CategoryService()
  //   await catService.create({
  //     id: 1,
  //     name: 'Mobilier',
  //     description:
  //       'Oat cake biscuit cookie chocolate pudding. Fruitcake cake cupcake bonbon pudding tiramisu jelly beans marzipan. Shortbread dessert chocolate biscuit jelly-o brownie. Toffee tootsie roll lollipop jelly beans powder croissant pie. Sesame snaps jelly-o dessert sugar plum donut caramels chocolate. Bear claw macaroon brownie cookie icing jelly sugar plum.',
  //     image:
  //       'https://www.highgest.fr/image/562535501368/1920/928/Les-petits-+.jpg',
  //   })

  //   await catService.create({
  //     id: 2,
  //     name: 'Eclairage',
  //     description:
  //       'Oat cake biscuit cookie chocolate pudding. Fruitcake cake cupcake bonbon pudding tiramisu jelly beans marzipan. Shortbread dessert chocolate biscuit jelly-o brownie. Toffee tootsie roll lollipop jelly beans powder croissant pie. Sesame snaps jelly-o dessert sugar plum donut caramels chocolate. Bear claw macaroon brownie cookie icing jelly sugar plum.',
  //     image:
  //       'https://www.highgest.fr/image/505818799390/1920/945/Les-petits-+.jpg',
  //   })

  //   const res = await execute(getCategories)
  //   expect(res).toMatchSnapshot()
  // })
  // it('can get one category by id', async () => {
  //   const cat3 = await Category.create({
  //     name: 'Sonorisation',
  //     description: 'Description de la catégorie Sonorisation',
  //     image: 'https://exemple.com/image_sonorisation.jpg',
  //   }).save()
  //   const res = await execute(getCategoryById, { categoryId: cat3.id })
  //   expect(res).toMatchSnapshot()
  // })
})
