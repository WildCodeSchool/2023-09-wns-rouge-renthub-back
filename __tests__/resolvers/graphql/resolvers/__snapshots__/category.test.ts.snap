// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Resolver - Snapshot should match snapshot 1`] = `
{
  "data": null,
  "errors": [
    [GraphQLError: No metadata for "Category" was found.],
  ],
}
`;
